### PR TITLE
clean up and reorg module-info.java

### PR DESCRIPTION
**Pull request by gavinking**
- some topics were very inadequately covered, considering their importance in Jakarta Data
- the organization of the material was quite confusing
- there was some outdated info

### CODE REVIEW

File: api/src/main/java/module-info.java [hunk 1]
```diff
@@ -157,7 +158,24 @@
  * }
  * </pre>
  *
- * <h2>Persistent Field Names</h2>
+ * <h2>Entities</h2>
+ *
+ * <p>Every entity programming model specifies an entity-defining annotation.
+ * For Jakarta Persistence, this is {@code jakarta.persistence.Entity}. For
+ * Jakarta NoSQL, it is {@code jakarta.nosql.Entity}. An entity class is
+ * identified by its entity-defining annotation.</p>
```
Comment (njr-11) [inline api/src/main/java/module-info.java:166]: This wasn't consistent with the spec document, so I adjusted it and copied over the language for unannotated
```suggestion
 * <p>An entity programming model typically specifies an entity-defining
 * annotation that is used to identify entity classes. For Jakarta Persistence,
 * this is {@code jakarta.persistence.Entity}. For Jakarta NoSQL, it is
 * {@code jakarta.nosql.Entity}. A provider may even have no entity-defining
 * annotation and feature a programming model for entity classes where the
 * entity classes are unannotated.</p>
```

**Review by njr-11 — CHANGES_REQUESTED** [review]

File: api/src/main/java/module-info.java [hunk 2]
```diff
@@ -549,115 +666,79 @@
...
+ *
+ * <p>Each parameter determines a query condition, and each such condition
+ * is an equality condition. All conditions must match in order that a record
+ * satisfy the query.</p>
+ *
+ * <pre>
+ * &#64;Find
+ * &#64;OrderBy("nationality")
+ * &#64;OrderBy("age")
+ * List&lt;Person&gt; peopleByAgeAndNationality(int age, Country nationality);
```
[the remark this answers is not in the document]
Comment (gavinking) [reply api/src/main/java/module-info.java:700]: Haha, very good point. How embarrassing :)

File: api/src/main/java/module-info.java (hunk 1 above)
Comment (gavinking) [reply api/src/main/java/module-info.java:166]: Yup, in attempting to make it more readable, I changed its meaning. Grr.

**Comment by gavinking** [comment]
@njr-11 I applied all your changes, which were all good catches.

**Review by njr-11 — APPROVED** [review]

**Review by otaviojava — APPROVED** [review]

**Pull request merged**